### PR TITLE
Support LLVM 14.0

### PR DESCRIPTION
**Pull request by HertzDevil**
LLVM 14.0 final is scheduled to be released on March 15th (overdue). This PR contains minimal changes to get it working, due to https://reviews.llvm.org/D108614 and https://reviews.llvm.org/D116014.

### CODE REVIEW

**Review by straight-shoota — APPROVED** [review]

File: src/llvm/lib_llvm.cr [hunk 1]
```diff
@@ -19,6 +19,7 @@ end
 
 {% begin %}
   lib LibLLVM
+    IS_140 = {{LibLLVM::VERSION.starts_with?("14.0")}}
```
Comment (Sija) [inline src/llvm/lib_llvm.cr:22]: I'd also add `IS_LT_140`.

File: .github/workflows/linux.yml [hunk 2]
```diff
@@ -64,6 +64,9 @@ jobs:
         - llvm_version: 13.0.0
           llvm_url: https://github.com/llvm/llvm-project/releases/download/llvmorg-13.0.0/clang+llvm-13.0.0-x86_64-linux-gnu-ubuntu-20.04.tar.xz
           base_image: ubuntu-20.04
+        - llvm_version: 14.0.0
+          llvm_url: https://github.com/llvm/llvm-project/releases/download/llvmorg-14.0.0-rc2/clang+llvm-14.0.0-rc2-x86_64-linux-gnu-ubuntu-18.04.tar.xz
```
[the remark this answers is not in the document]
Comment (HertzDevil) [reply .github/workflows/linux.yml:68]: No official pre-built binaries are up yet, not even rc4

File: .github/workflows/linux.yml [hunk 3]
```diff
@@ -64,6 +64,9 @@ jobs:
         - llvm_version: 13.0.0
           llvm_url: https://github.com/llvm/llvm-project/releases/download/llvmorg-13.0.0/clang+llvm-13.0.0-x86_64-linux-gnu-ubuntu-20.04.tar.xz
           base_image: ubuntu-20.04
+        - llvm_version: 14.0.0
+          llvm_url: https://github.com/llvm/llvm-project/releases/download/llvmorg-14.0.0/clang+llvm-14.0.0-x86_64-linux-gnu-ubuntu-18.04.tar.xz
+          base_image: ubuntu-18.04
```
Comment (straight-shoota) [inline .github/workflows/linux.yml:69]: Is there a reason for using this old image?

File: src/llvm/lib_llvm.cr (hunk 1 above)
Comment (straight-shoota) [reply src/llvm/lib_llvm.cr:22]: Apparently there's no need for that, so why bother?

**Review by beta-ziliani — APPROVED** [review]
As always, thank you very much HertzDevil 🙇

**Pull request merged**